hybrid_robotV0_2:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Position Controllers ---------------------------------------
  back_left_wheel_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: back_left_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  front_left_wheel_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: front_left_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  back_right_wheel_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: back_right_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  front_right_wheel_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: front_right_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  extension_base_joint_position_controller:
    type: effort_controllers/JointPositionController
    joint: extension_base_joint
    pid: { p: 1000.0, i: 0.1, d: 100.0 }

  back_left_wheel_joint_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: back_left_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  front_left_wheel_joint_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: front_left_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  back_right_wheel_joint_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: back_right_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  front_right_wheel_joint_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: front_right_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  extension_base_joint_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: extension_base_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }

  back_left_wheel_joint_effort_controller:
    type: effort_controllers/JointEffortController
    joint: back_left_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  front_left_wheel_joint_effort_controller:
    type: effort_controllers/JointEffortController
    joint: front_left_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  back_right_wheel_joint_effort_controller:
    type: effort_controllers/JointEffortController
    joint: back_right_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  front_right_wheel_joint_effort_controller:
    type: effort_controllers/JointEffortController
    joint: front_right_wheel_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
  extension_base_joint_effort_controller:
    type: effort_controllers/JointEffortController
    joint: extension_base_joint
    pid: { p: 100.0, i: 0.01, d: 10.0 }
